WordPress/Classes/Services/SharingService.swift

```diff
@@ -451,25 +455,20 @@ public class SharingService : LocalCoreDataService
     // MARK : Private Instance Methods
 
 
-    /// Returns the API to use with the service
+    /// Returns the API to use with the service. 
     ///
-    private func apiForRequest() -> WordPressComApi {
-        var api: WordPressComApi? = nil
-
-        if let restApi = AccountService(managedObjectContext: managedObjectContext).defaultWordPressComAccount()?.restApi {
-            api = restApi.hasCredentials() ? restApi : nil
-        }
-
-        if api == nil {
-            // In practice this should never happen, but if it does let's try to detect it. 
-            // Write to the error log if the api was nil, and trigger an assert to 
-            // catch this in development/QA.
-            api = WordPressComApi.anonymousApi()
-            let error = "SharingService is not using a real WordPress.com account."
-            DDLogSwift.logError(error)
-            assert(false, error)
+    /// - Note:
+    ///     An anonymous API is returned for self-hosted blog not connected to Jetpack.
+    ///
+    /// - Parameters:
+    ///     - blog: The blog to use for the rest api.
+    ///
+    private func apiForBlog(blog: Blog) -> WordPressComApi {
+        if let api = blog.restApi() {
+            return api
         }
 
-        return api!
+        DDLogSwift.logError("SharingService is not using a real WordPress.com account.")
+        return WordPressComApi.anonymousApi()
```

Comment:
Yep.  The assert doesn't make sense when we're intending to return the anonymous api. However, that `logError` should probably be a `logDebug` with a better message.